includes/backends/class-omise-backend-installment.php

```diff
@@ -0,0 +1,123 @@
+	/**
+	 * @var array  of known installment providers.
+	 */
+	protected static $providers = array();
+
+	public function initiate() {
+		self::$providers = array(
+			'installment_first_choice' => array(
+				'bank_code'          => 'first_choice',
+				'title'              => __( 'Krungsri First Choice', 'omise' ),
+				'interest_rate'      => 0.013,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_bay' => array(
+				'bank_code'          => 'bay',
+				'title'              => __( 'Krungsri', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_ktc' => array(
+				'bank_code'          => 'ktc',
+				'title'              => __( 'Krungthai Card (KTC)', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_bbl' => array(
+				'bank_code'          => 'bbl',
+				'title'              => __( 'Bangkok Bank', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 500.00,
+			),
+
+			'installment_kbank' => array(
+				'bank_code'          => 'kbank',
+				'title'              => __( 'Kasikorn Bank', 'omise' ),
+				'interest_rate'      => 0.0065,
+				'min_allowed_amount' => 500.00,
+			),
+		);
+	}
+
+	/**
+	 * @param  string $currency
+	 * @param  float  $purchase_amount
+	 *
+	 * @return array  of an available installment providers
+	 */
+	public function get_available_providers( $currency, $purchase_amount ) {
+		// Note: as installment payment at the moment only supports for THB currency, so the 
+		//       $purchase_amount is multiplied with 100 to convert the amount into subunit (satang).
+		$providers = $this->capabilities()->getInstallmentBackends( $currency, ( $purchase_amount * 100 ) );
+
+		foreach ( $providers as &$provider ) {
+			$provider->provider_code             = str_replace( 'installment_', '', $provider->_id );
+			$provider->provider_name             = isset( self::$providers[ $provider->_id ] ) ? self::$providers[ $provider->_id ]['title'] : strtoupper( $provider->code );
+			$provider->allowed_installment_terms = $this->get_valid_terms( $provider->allowed_installment_terms, self::$providers[ $provider->_id ], $purchase_amount );
+			$provider->interest                  = $this->capabilities()->is_zero_interest() ? 0 : ( self::$providers[ $provider->_id ]['interest_rate'] * 100 );
+		}
+
+		return $providers;
+	}
+
+	/**
+	 * @param  array $available_terms
+	 * @param  array $provider_detail
+	 * @param  float $purchase_amount
+	 *
+	 * @return array  of an filtered available terms
+	 */
+	public function get_valid_terms( $available_terms, $provider_detail, $purchase_amount ) {
+		$valid_terms = array();
+
+		sort( $available_terms );
+
+		foreach ( $available_terms as $term ) {
+			$monthly_amount = $this->calculate_monthly_payment_amount(
+				$purchase_amount,
+				$term,
+				$this->capabilities()->is_zero_interest() ? 0 : $provider_detail['interest_rate']
```

Comment:
Wasn't this `is_zero_interest` check already done in `get_available_providers`?

Reply:
@jonrandy Good point 👍  I think I can clean this one a bit more